galley/pkg/config/processing/snapshotter/analyzingdistributor.go

```diff
@@ -46,13 +49,20 @@ const syntheticSnapshotGroup = "syntheticServiceEntry"
 // NewAnalyzingDistributor returns a new instance of AnalyzingDistributor.
 func NewAnalyzingDistributor(u StatusUpdater, a analysis.Analyzer, d Distributor) *AnalyzingDistributor {
 	return &AnalyzingDistributor{
-		updater:       u,
-		analyzer:      a,
-		distributor:   d,
-		lastSnapshots: make(map[string]*Snapshot),
+		updater:            u,
+		analyzer:           a,
+		distributor:        d,
+		lastSnapshots:      make(map[string]*Snapshot),
+		collectionReporter: func(collection.Name) {}, // No-op unless SetCollectionReporter specifies otherwise
 	}
 }
 
+// SetCollectionReporter adds a hook that gets called whenever a collection is accessed
+// through the AnalyzingDistributor's context.
+func (d *AnalyzingDistributor) SetCollectionReporter(fn func(collection.Name)) {
```

Comment:
Make this a parameter to NewAnalyzingDistributor? This should not be changed during runtime, right?

Reply:
The reasoning for doing it this way was to allow a default (no-op) value for the hook and to keep the scope of the change down. If go had support for default values in function arguments, I would have used that.

Went ahead and changed it such that if you pass in nil, it uses a no-op default.